File: app/src/main/java/com/juawapps/newsspread/di/news/NewsFragmentBuildersModule.java
package com.juawapps.newsspread.di.news;

import com.juawapps.newsspread.ui.news.NewsFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * News fragment builders.
 */

@SuppressWarnings("unused")
@Module
public abstract class NewsFragmentBuildersModule {

    @ContributesAndroidInjector
    abstract NewsFragment contributeNewsFragment();
}

File: app/src/main/java/com/juawapps/newsspread/di/news/NewsFragmentBuildersModule.kt
package com.juawapps.newsspread.di.news

import com.juawapps.newsspread.ui.news.NewsFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * News fragment builders.
 */

@Module
abstract class NewsFragmentBuildersModule {

    @ContributesAndroidInjector(modules = arrayOf(NewsFragmentModule::class))
    internal abstract fun contributeNewsFragment(): NewsFragment
}
